src/services/PostCache/findPostsInCache.ts

```diff
@@ -0,0 +1,56 @@
+import PostCache from "../redisCache";
+import type { InterfacePost } from "../../models";
+import { Types } from "mongoose";
+
+export async function findPostsInCache(
+  ids: string[] | any[]
+): Promise<(InterfacePost | null)[]> {
+  const keys: string[] = ids.map((id) => {
+    return `post:${id}`;
+  });
+
+  const postsFoundInCache = await PostCache.mget(keys);
+
+  const posts = postsFoundInCache.map((post) => {
+    if (post === null) {
+      return null;
+    }
+
+    try {
+      const postObj = JSON.parse(post);
+
+      // Note: While JSON parsing successfully restores the fields, including those with
+      // Mongoose Object IDs, these fields are returned as strings due to the serialization
+      // process. To ensure accurate data representation, we manually convert these string
+      // values back to their original Mongoose Object ID types before delivering them to
+      // the requesting resolver.
+
+      return {
```

Comment:
Not sure I understand by this was done like this

Reply:
Yeah so basically in the cache we are storing our data in JSON format as values, the problem is when we parse the JSON object after fetching a post, comment, org , event etc the values for fields after parsing an object are all converted to primitive data types. 

So in the resolvers/model there are certain types like Mongoose Object ID or JS date-time type which are all converted to string values. This was causing the tests to fail due to runtime errors and that's why this extra step was needed.

Reply:
Noted